src/details/ArborX_AccessTraits.hpp

```diff
@@ -206,6 +208,10 @@ struct AccessTraits<
         AccessTraitsNotSpecializedArchetypeAlias, Traits::Access<T, Tag>>{}>>
     : Traits::Access<T, Tag>
 {
+  static_assert(
+      KokkosExt::is_detected<AccessTraitsNotSpecializedArchetypeAlias,
+                             Traits::Access<T, Tag>>{},
```

Comment:
I would have thought this should be `!KokkosExt::...`, but it works correctly without `!`.